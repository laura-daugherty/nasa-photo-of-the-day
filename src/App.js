import React, { useState, useEffect } from "react";
import "./App.css";
import axios from "axios"
import Hero from "./components/hero"
import Header from "./components/header"

import { setUseProxies } from "immer";

function App() {
  const [data, setData] = useState("");
  const [picture, setPicture] = useState("");
  // const [loading, setLoading] = useState();
  // const [error, setError] = useState("");

  // useEffect(() => {
  //   fetchData()
  //     .then(result => setData(result.data))
  //     .catch(error => setError("It's an Error"))

  // }, []);

  useEffect(() => {
    console.log("useEffect fetching Data")
    axios
      .get("https://api.nasa.gov/planetary/apod?api_key=DEMO_KEY&date=2012-03-04")
      .then(result => setData(result.data))
        //where do I console.log() to see the data??
      .catch(err => console.log(err));
  }, [] );

  return (
    <div className="App">
      <Header />
      <Hero 
        data={data} 
      />
    </div>
  );
}

export default App;
