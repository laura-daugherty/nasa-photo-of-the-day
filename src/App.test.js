import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'

import App from './App';
import Hero from "./components/hero"
import Header from "./components/header"

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('<App />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<Hero />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<Hero />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<Heading />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<Header />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

