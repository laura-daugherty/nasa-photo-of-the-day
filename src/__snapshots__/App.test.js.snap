// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="App"
>
  <header>
    <img
      src="public/nasa-logo-web-rgb.png"
    />
    <h2
      className="ui center aligned icon header"
    >
      Nasa is cool
    </h2>
  </header>
  <div
    class="ui active inverted dimmer"
  >
    <div
      class="ui large text loader"
      className="ui text loader"
    >
      Fetching Your Photo
    </div>
  </div>
</div>
`;

exports[`<Heading /> should match snapshot 1`] = `
<header>
  <img
    src="public/nasa-logo-web-rgb.png"
  />
  <h2
    className="ui center aligned icon header"
  >
    Nasa is cool
  </h2>
</header>
`;

exports[`<Hero /> should match snapshot 1`] = `
<div
  class="ui active inverted dimmer"
>
  <div
    class="ui large text loader"
    className="ui text loader"
  >
    Fetching Your Photo
  </div>
</div>
`;
